target/linux/ath79/image/nand.mk

```diff
@@ -77,6 +90,41 @@ define Device/domywifi_dw33d
 endef
 TARGET_DEVICES += domywifi_dw33d
 
+define Device/dongwon_dw02-412h
+  SOC := qca9557
+  DEVICE_VENDOR := Dongwon T&I
+  DEVICE_MODEL := DW02-412H
+  DEVICE_ALT0_VENDOR := KT
+  DEVICE_ALT0_MODEL := GiGA WiFi home
+  DEVICE_PACKAGES := kmod-usb2 kmod-ath10k-ct ath10k-firmware-qca988x-ct
+  KERNEL_SIZE := 8192k
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma | dw-headers
+  KERNEL_INITRAMFS := kernel-bin | append-dtb | lzma | uImage lzma | dw-headers
+  UBINIZE_OPTS := -E 5
+  IMAGES += factory.img
+  IMAGE/factory.img := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi | \
+	check-size
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+endef
+
+define Device/dongwon_dw02-412h-64m
+  $(Device/dongwon_dw02-412h)
+  DEVICE_VARIANT := (64M)
+  DEVICE_ALT0_VARIANT := (DW02-412H) (64M)
```

Comment:
Why put the name in brackets another time?

Reply:
It was a suggestion by @mans0n ([link](https://github.com/openwrt/openwrt/pull/3930#discussion_r585487803)). If you think this isn't right, I'll remove it.

Reply:
I thought what Adrian meant was something like `DEVICE_ALT0_VARIANT := (DW02-412H 64M)`...

Reply:
To be clear,
1. There are three "GiGA WiFi home" devices, DW02-412H, KM06-506H, KM06-704H. Users might be confused if the submodel is not specified.
2. The autocomplete search box of [firmware-selector](https://firmware-selector.openwrt.org/) does not append original name after alt name, so 64/128M images would not be distinguishable at a glance.

Reply:
What I meant was I don't see why we need "(DW02-412H)" in variant another time ...

Reply:
Sorry, I misunderstood your words, but don't you think that alt name should be some kind of unique identifier by itself? Or do you think it is just for additional info appended to the original name?